components/script/lib.rs

```diff
@@ -199,6 +202,21 @@ pub fn init_service_workers(sw_senders: SWManagerSenders) {
     ServiceWorkerManager::spawn_manager(sw_senders);
 }
 
+#[no_mangle]
+#[allow(unsafe_code)]
+pub unsafe extern "C" fn MemoryReporter(obj: *mut JSObject) -> bool {
+    if obj.is_null() {
+        return false;
+    }
+    if is_platform_object(obj) || is_dom_proxy(obj) {
```

Comment:
This can be simplified to:
```rust
pub unsafe extern "C" fn MemoryReporter(obj: *mut JSObject) -> bool {
    !obj.is_null() && (is_platform_object(obj) || is_dom_proxy(obj))
}
```